core/src/main/java/com/linecorp/armeria/client/ClientFactory.java

```diff
@@ -369,4 +371,14 @@ default ClientBuilderParams validateParams(ClientBuilderParams params) {
         }
         return params;
     }
+
+    /**
+     * Add a shutdown hook to stop this {@link ClientFactory}.
+     */
+    default void closeOnShutdown() {
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            closeDefault();
+            logger.info("ClientFactory has been closed.");
```

Comment:
Probably OK to omit this log message? or we could log at DEBUG level.

Reply:
I'd like to set debug level to this.